include/boost/rational.hpp

```diff
@@ -671,6 +690,22 @@ inline rational<IntType> abs(const rational<IntType>& r)
     return r.numerator() >= IntType(0)? r: -r;
 }
 
+template <typename IntType>
+inline rational<IntType> pow(rational<IntType> const& base, IntType exponent)
+{
+    if (!base) return base;
+    bool const positive = exponent >= 0;
+    rational<IntType> result(exponent % 2 != 0? base: rational<IntType>(1));
+    rational<IntType> x(base);
+
```

Comment:
It would be better to remove the `const&` from the parameter `base` and just use that, rather than copy it into `x` here.

Reply:
@kundor in that case we would just be moving the copy to the caller; it seems more correct (as it is currently stated) to claim that base will not be modified by the call.